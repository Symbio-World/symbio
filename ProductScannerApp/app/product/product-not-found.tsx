import React from 'react'
import { View, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { Button } from '../ui-kit/button'

type Props = {
  barcode?: string
}
export const ProductNotFound: React.FC<Props> = ({ barcode }) => {
  return (
    <View style={[t.flex1, t.justifyCenter]}>
      <Text>Unfortunately we could not found any product with this barcode<Text>{barcode}</Text></Text>
      <Button title='Add Manually' />
    </View>
  )
}
