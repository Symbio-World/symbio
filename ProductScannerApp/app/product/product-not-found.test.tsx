import 'react-native'
import React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import { ProductNotFound } from './product-not-found'

describe('ProductNotFound', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<ProductNotFound />)
    expect(toJSON()).toMatchSnapshot()
  })
})
