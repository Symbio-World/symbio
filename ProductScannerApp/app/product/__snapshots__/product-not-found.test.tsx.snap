// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductNotFound renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <Text>
    Unfortunately we could not found any product with this barcode
    <Text />
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#48bb78",
        "borderRadius": 4,
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "padding": 16,
        "textAlignVertical": "center",
      }
    }
    testID="setup-tags-screen-submit"
  >
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 24,
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Add Manually
    </Text>
  </View>
</View>
`;
